src/components/Editor/tests/Editor.spec.js

```diff
@@ -42,38 +29,123 @@ function render(overrides = {}) {
     replaceDocument: jest.fn(),
     setMode: jest.fn()
   };
-  Editor.WrappedComponent.prototype.setupEditor = () => {
-    return mockEditor;
-  };
+}
+
+function createMockSource(overrides) {
+  return I.fromJS({
+    id: "foo",
+    text: "the text",
+    loadedState: "loaded",
+    url: "foo",
+    ...overrides
+  });
+}
+
+function render(overrides = {}) {
+  const props = generateDefaults(overrides);
+  const mockEditor = createMockEditor();
   const component = shallow(<Editor.WrappedComponent {...props} />, {
     context: {
       shortcuts: { on: jest.fn() }
     }
   });
+
   return { component, props, mockEditor };
 }
 
 describe("Editor", () => {
-  it("should render", async () => {
-    const { component } = render();
-    expect(component).toMatchSnapshot();
+  describe("When empty", () => {
+    it("should render", async () => {
+      const { component } = render();
+      expect(component).toMatchSnapshot();
+    });
+  });
+
+  describe("When loading initial source", () => {
+    it("should show a loading message", async () => {
+      const { component, mockEditor } = render();
+      await component.setState({ editor: mockEditor });
+      component.setProps({
+        selectedSource: I.fromJS({ loadedState: "loading" })
+      });
+
+      expect(mockEditor.setText.mock.calls).toEqual([["Loading…"]]);
+      expect(mockEditor.codeMirror.scrollTo.mock.calls).toEqual([]);
+    });
   });
 
-  it("should be loading", async () => {
-    const { component, mockEditor } = render();
-    await component.setState({ editor: mockEditor });
-    component.setProps({
-      selectedSource: I.fromJS({ loadedState: "loading" })
+  describe("When loaded", () => {
+    it("should show text", async () => {
+      const { component, mockEditor, props } = render({});
+
+      await component.setState({ editor: mockEditor });
+      await component.setProps({
+        ...props,
+        selectedSource: createMockSource({ loadedState: "loaded" }),
+        selectedLocation: { sourceId: "foo", line: 3, column: 1 }
```

Comment:
would be better to get the mock values out so that we can test against them, and its clearer when reading the tests what is happening